src/Orchard.Web/Modules/Orchard.Users/Services/PasswordChangedDateUserDataProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using Orchard.ContentManagement;
+using Orchard.Security;
+using Orchard.Security.Providers;
+using Orchard.Settings;
+using Orchard.Users.Models;
+
+namespace Orchard.Users.Services {
+    public class PasswordChangedDateUserDataProvider : BaseUserDataProvider {
```

Comment:
Does it need to be extensible? Isn't the setting already available in the form authentication service?

Reply:
Could you clarify what you mean? I don't understand what you are referring to.

Reply:
I don't see why we need to be able to extend this. The forms auth logic should be self sufficient, and store in the auth cookie only what it needs. If other modules need to store stuff for the user in a cookie, they are free to create their own.

Reply:
The point is not to store information in the auth cookie per se, but to stamp it with information to be able to invalidate it if/when required by something. In the specific case of PasswordChangedDateUserDataProvider, the idea is that a password change can make all auth cookies, for the user and from any client, invalid. That information we are storing in the cookie (the datetime in this case) is not for the user/client.